bdk_core/src/spk_tracker.rs

```diff
@@ -39,51 +37,32 @@ impl<I> Default for SpkTracker<I> {
             spk_indexes: Default::default(),
             spk_txouts: Default::default(),
             unused: Default::default(),
-            txid_digests_seen: Default::default(),
         }
     }
 }
 
 impl<I: Clone + Ord> SpkTracker<I> {
-    pub fn sync(&mut self, chain: &SparseChain) {
-        // by reverse iterating the checkpoints and collecting the transactions until we find a
-        // checkpoint with a txid digest we've seen we avoid having to apply *all* the transactions
-        // in the sparse chain every time.
-        let txids_to_add = chain
-            .iter_checkpoints(..)
-            .rev()
-            .take_while(|checkpoint_id| {
-                !self
-                    .txid_digests_seen
-                    .contains(&chain.txid_digest_at(*checkpoint_id))
-            })
-            .flat_map(|checkpoint_id| chain.checkpoint_txids(checkpoint_id))
-            .chain(chain.unconfirmed().iter().cloned())
-            .collect::<Vec<_>>();
-
-        for txid_to_add in txids_to_add {
-            self.add_tx(txid_to_add, chain);
-        }
-
-        if let Some(latest_checkpoint) = chain.latest_checkpoint() {
-            let latest_txid_digest = chain.txid_digest_at(latest_checkpoint);
-            self.txid_digests_seen.insert(latest_txid_digest);
-        }
+    pub fn sync(&mut self, chain: &SparseChain, graph: &TxGraph) {
+        chain
+            .iter_txids()
+            .map(|(_, txid)| graph.tx(&txid).expect("tx must exist"))
+            .for_each(|tx| self.add_tx(tx));
```

Comment:
I think we can just iter txs on the `graph` no need to pass in `chain`. Actually it would be good to be able to have a function with `iter_txout` on `graph` so when we add auxiliary txouts we can add them to the tracker without changing code here.

Reply:
Done.